pipeline/backend/kubernetes/pod.go

```diff
@@ -113,17 +120,49 @@ func stepLabel(step *types.Step) (string, error) {
 	return toDNSName(step.Name)
 }
 
-func podSpec(step *types.Step, config *config) (v1.PodSpec, error) {
+func podSpec(step *types.Step, config *config, labels map[string]string) (v1.PodSpec, error) {
 	var err error
-	spec := v1.PodSpec{
-		RestartPolicy:      v1.RestartPolicyNever,
-		ServiceAccountName: step.BackendOptions.Kubernetes.ServiceAccountName,
-		ImagePullSecrets:   imagePullSecretsReferences(config.ImagePullSecretNames),
-		HostAliases:        hostAliases(step.ExtraHosts),
-		NodeSelector:       nodeSelector(step.BackendOptions.Kubernetes.NodeSelector, step.Environment["CI_SYSTEM_PLATFORM"]),
-		Tolerations:        tolerations(step.BackendOptions.Kubernetes.Tolerations),
-		SecurityContext:    podSecurityContext(step.BackendOptions.Kubernetes.SecurityContext, config.SecurityContext),
+	spec := v1.PodSpec{}
+
+	repoLabel, hasLabelRepo := labels["repository"]
+	pipelineNumberLabel, hasLabelPipelineNumber := labels["pipeline_number"]
+
+	// force service and steps to run in the same node when rwx is false
+	// maybe in the future get theses values from backend configuration
+	if !config.StorageRwx && hasLabelRepo && hasLabelPipelineNumber {
```

Comment:
we should have tests for that :)

Reply:
done, do you see any other test we need to do?